### PR TITLE
✨Revoke Allowances

### LINKED ISSUE #1193: Allowance information + revoke/change input and button
Related to #1190

There should be a section in the app that shows all current allowances where the spender is one of our contracts. And provide the user a way to revoke or update those amounts.

### PR DESCRIPTION
**Pull request by sebipap**
closes #1193 

### Manage allowances for 
* DebtManager => (`USDC`, `OP`, `WETH`, ...)
* DebtManager => market shares (`exaUSDC`, `exaOP` ...)
* Markets =>  `USDC`, `OP`, `WETH`, ...
* MarketETHRouter => `exaWETH`

### Note for testing
* Manage allowances in the wallet address menu in the top-right of the screen

image

### CODE REVIEW

File: components/Allowances/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    title: i18n.t('Authorized Amount'),
+    sortKey: 'allowanceUSD',
+    DisplayComponent: Amount,
+  },
+  {
+    title: i18n.t('Authorized Spender'),
+    sortKey: 'spenderName',
+    DisplayComponent: Spender,
+  },
+] as const;
```
Comment (jgalat) [inline components/Allowances/index.tsx:28]: using the global `i18n.t` won't re-compute translations if the user changes the lang.
You can just wrap this obj in a func, or just use it inside a component and use `{ t }` from `useTranslation`.

**Review by jgalat — APPROVED** [review]
lgtm! great work, just a minor comment

File: components/Allowances/index.tsx (hunk 1 above)
Comment (sebipap) [reply components/Allowances/index.tsx]: fixed, good catch!

**Review by jgalat — DISMISSED** [review]

**Comment by sebipap** [comment]
> Nice work Seba! Just a few comments regarding the UI:
> 1. The menu width should be smaller
> 2. The button icons should be smaller
> 3. The copy button should be bigger
> 4. The divider line is missing
thanks for the feedback juampi!

1,2,4 ✅
3: I didn't change this because it was not the scope of the change (only the below buttons where changed) and also is a common component used across the app and it's better to mantain consistency IMO.

File: components/Allowances/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        </Box>
+        <Box bgcolor="components.bg" borderRadius={2} gap={2} boxShadow={'0px 3px 4px 0px rgba(97, 102, 107, 0.10)'}>
+          <AllowancesComponent data={data} loading={loading} update={update} />
+        </Box>
+      </Box>
+    </Box>
+  );
+};
+
+export default memo(Allowances);
```
Comment (JuampiRombola) [inline components/Allowances/index.tsx:63]: `nit`: memo is not necessary as it has no props

File: pages/revoke.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Allowances from 'components/Allowances';
+import { usePageView } from 'hooks/useAnalytics';
+import React, { memo } from 'react';
+
+const Revoke = () => {
+  usePageView('/revoke', 'Manage Allowances');
+  return <Allowances />;
+};
+
+export default memo(Revoke);
```
Comment (JuampiRombola) [inline pages/revoke.tsx:10]: `nit`: memo is not necessary as it has no props

**Review by JuampiRombola — APPROVED** [review]
lgtm

**Comment by JuampiRombola** [comment]
Another comment (if you consider it is in the scope of your task), could you reduce the gap between the menu and the navbar?
image
image

File: components/Allowances/index.tsx (hunk 2 above)
Comment (sebipap) [reply components/Allowances/index.tsx:63]: Isn't it the other way around? `React.memo()` is actually worse for component with props, specially with recurrently changing props. The best use of `memo` is for components that don't take props but have costly renders.
Although I'm not sure and not an expert on this.

File: pages/revoke.tsx (hunk 3 above)
Comment (sebipap) [reply pages/revoke.tsx:10]: same as above

File: components/Allowances/index.tsx (hunk 2 above)
Comment (JuampiRombola) [reply components/Allowances/index.tsx:63]: "**memo** lets you skip re-rendering a component when its props are unchanged.", https://react.dev/reference/react/memo

Comment (sebipap) [reply components/Allowances/index.tsx:63]: yes, but if no props, the component doesnt re-render. both are `undefined`

**Comment by sebipap** [comment]
> Another comment (if you consider it is in the scope of your task), could you reduce the gap between the menu and the navbar? 
fixed!

File: components/Allowances/index.tsx (hunk 2 above)
Comment (JuampiRombola) [reply components/Allowances/index.tsx:77]: you're correct; that's why I said it is not necessary

File: components/Wallet/index.tsx [hunk 4]
```diff
@@ -129,17 +131,32 @@ function Wallet() {
...
-            onClick={() => {
-              closeMenu();
-              impersonateActive ? exitImpersonate() : disconnect();
-            }}
-            sx={{ color: 'grey.700', borderColor: '#CFD3D8' }}
-          >
-            {impersonateActive ? t('Exit Read-Only Mode') : t('Disconnect')}
-          </Button>
+          <List disablePadding sx={{ borderTop: 1, borderColor: 'grey.300', pt: 1, px: 1 }}>
+            <Link href="/revoke">
```
Comment (JuampiRombola) [inline components/Wallet/index.tsx:135]:
```suggestion
            <Link href={{ pathname: 'revoke', query }} legacyBehavior>
```

File: components/Allowances/Spender.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+const Spender = ({ spenderName, spenderAddress }: Pick<Allowance, 'spenderName' | 'spenderAddress'>) => {
+  const { address } = useEtherscanLink();
+  return (
+    <>
+      <Typography fontSize={19} fontWeight={500}>
+        {spenderName}
+      </Typography>
+      <Typography fontFamily="IBM Plex Mono" fontSize={14} fontWeight={500} color="grey.500">
+        <Stack direction="row">
+          <Link href={address(spenderAddress)}>
```
Comment (JuampiRombola) [inline components/Allowances/Spender.tsx:18]:
```suggestion
          <Link href={address(spenderAddress)} target="_blank" rel="noopener noreferrer">
```

File: components/Allowances/Asset.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            maxWidth: '100%',
+            height: 'auto',
+          }}
+        />
+        <Typography display="inline" alignSelf="center" fontSize={19} fontWeight={500} ml={1}>
+          {symbol}
+        </Typography>
+      </Stack>
+      <Stack direction="row">
+        <Link href={address(token)}>
```
Comment (JuampiRombola) [inline components/Allowances/Asset.tsx:30]:
```suggestion
        <Link href={address(token)}  target="_blank" rel="noopener noreferrer">
```

Comment (sebipap) [reply components/Allowances/Asset.tsx]: fixed

File: components/Allowances/Spender.tsx (hunk 5 above)
Comment (sebipap) [reply components/Allowances/Spender.tsx]: fixed

File: components/Wallet/index.tsx (hunk 4 above)
Comment (sebipap) [reply components/Wallet/index.tsx]: fixed

File: components/Wallet/index.tsx [hunk 7]
```diff
@@ -1,15 +1,19 @@
 import React, { useMemo } from 'react';
+import { useRouter } from 'next/router';
```
Comment (JuampiRombola) [inline components/Wallet/index.tsx:2]:
```suggestion
import useRouter from 'hooks/useRouter';
```

**Review by JuampiRombola — DISMISSED** [review]

**Comment by jgalat** [comment]
This is getting merged as part of #1201

**Pull request closed without merging**